2.75

ATSchoolENikonov opened the pull request:
Что-то вышло, удаления фрукта не использовал, так как у Листа уже есть метод удаления)

methshest committed:
2.75
Что-то вышло, удаления фрукта не использвоал, так как у Листа уже есть метод удаления)

Separador commented on 2.58_homework/com/fruitbase/FruitBase.java:
```diff
@@ -1,5 +1,17 @@
 package com.fruitbase;
 
+import com.fruitbase.Customers.Customer;
+import com.fruitbase.Customers.FreshCustomer;
+import com.fruitbase.Customers.UniqueCustomer;
+import com.fruitbase.fruits.Apple;
+import com.fruitbase.fruits.Banana;
+import com.fruitbase.fruits.Fruit;
+import com.fruitbase.fruits.Pineapple;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
```
Подчисти неиспользуемые импорты.

Separador commented on 2.58_homework/com/fruitbase/fruits/Fruit.java:
```diff
@@ -25,8 +28,26 @@ public String getName() {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Fruit fruit = (Fruit) o;
+        return Double.compare(fruit.weight, weight) == 0 && freshness == fruit.freshness && price.equals(fruit.price) && name.equals(fruit.name);
```
Строчки длиннее 100 символов разбивай на несколько

Separador commented on 2.58_homework/com/fruitbase/Customers/FreshCustomer.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Override
+    public List<Fruit> takeFruits(Cargo cargo) {
+        for (int i = 0; i < cargo.getFruit().size(); i++) {
+            if (cargo.getFruit().get(i).isFresh()) {
+                purchases.add(cargo.getFruit().get(i));
+            }
+        }
```
Для читаемости и отладки лучше сделай отдельную переменную, в которую будет записан результат `cargo.getFruit()` и ее используй.

Separador commented on 2.58_homework/com/fruitbase/Customers/UniqueCustomer.java:
```diff
@@ -0,0 +1,33 @@
+    @Override
+    public List<Fruit> takeFruits(Cargo cargo) {
+        int x = 0;
+        while (x < cargo.getFruit().size()) {
+            for (int i = x + 1; i < cargo.getFruit().size(); i++) {
+                if (cargo.getFruit().get(x).getName().equals(cargo.getFruit().get(i).getName())) {
+                    cargo.getFruit().remove(i);
+                }
+            }
+            purchases.add(cargo.getFruit().get(x));
```
Для читаемости и отладки лучше сделай отдельную переменную, в которую будет записан результат cargo.getFruit() и ее используй.

Separador commented on 2.58_homework/com/fruitbase/Simulation.java:
```diff
@@ -0,0 +1,30 @@
+        FruitBase fr = new FruitBase();
+        Customer[] customers = new Customer[]{new FreshCustomer(new ArrayList<>(), "Ivan"),
+                new UniqueCustomer(new ArrayList<>(), "Lexa")};
+        if (args.length != 0) {
+            Cargo car = new Cargo();
+            fr.takeOrder(args, car);
+            System.out.println(car);
+            System.out.println("Total Price: " + car.getPrice());
+            System.out.println("Total Weight: " + car.getWeight());
+            for (int i = 0; i < customers.length; i++) {
+                customers[i].takeFruits(car);
+            }
+            System.out.println(Arrays.toString(customers));
+        } else {
+            System.out.println("No valid order");
+            System.exit(12);
+        }
```
Для читаемости лучше сделай отдельный `if (args.length == 0)` с ранним выходом вместо ветки `else`. И тогда код из текущей основной ветки будет без отступа.

Separador commented on 2.58_homework/com/fruitbase/Simulation.java:
```diff
@@ -0,0 +1,30 @@
+public class Simulation {
+    public static void main(String[] args) {
+        FruitBase fr = new FruitBase();
+        Customer[] customers = new Customer[]{new FreshCustomer(new ArrayList<>(), "Ivan"),
+                new UniqueCustomer(new ArrayList<>(), "Lexa")};
```
Лучше перенеси как 
```
new Customer[] {
    new FreshCustomer(new ArrayList<>(), "Ivan"),
    new UniqueCustomer(new ArrayList<>(), "Lexa")
};
```

Separador reviewed (changes requested):
Хорошо, давай поправим мелочи и будем засчитывать задание

methshest committed:
2.75
Что-то вышло, удаления фрукта не использвоал, так как у Листа уже есть метод удаления)

Separador commented on 2.58_homework/com/fruitbase/Customers/UniqueCustomer.java:
```diff
@@ -12,14 +12,15 @@ public UniqueCustomer(List<Fruit> purchases, String name) {
-                if (cargo.getFruit().get(x).getName().equals(cargo.getFruit().get(i).getName())) {
-                    cargo.getFruit().remove(i);
+            for (int i = x + 1; i < k.size(); i++) {
+                if (k.get(x).getName().equals(k.get(i).getName())) {
```
Пропустил момент. Тут надо просто equals объектов, а не только по именам, тк как минимум они могут отличаться freshness.

methshest committed:
2.75
Что-то вышло, удаления фрукта не использвоал, так как у Листа уже есть метод удаления)
